tests/test_deprecations.py

```diff
@@ -284,3 +285,20 @@ def test_migrated_metadata_attribute():
             NumericalDiscreteParameter("p", [0, 1])
         )
         subspace.metadata
+
+
+@pytest.mark.parametrize(
+    "flag",
+    [
+        "allow_repeated_recommendations",
+        "allow_recommending_already_measured",
+        "allow_recommending_pending_experiments",
+    ],
+)
+def test_migrated_allow_flags(flag):
+    """Passing and accessing the migrated 'allow_*' flags raises an error."""
+    with pytest.raises(DeprecationError, match=r"Passing 'allow_\*' flags"):
+        RandomRecommender(**{flag: True})
```

Comment:
I think one of them would also suffice since stuff is defined right in the base class but because the test overhead is minimal and it's a deprecation test (i.e. it'll disappear anyway soon) I'll not be to stingy 🙃 Added.